Form submit changes

alexandercyu opened the pull request:
- Prevents default form submission from the HTML submit function
- Adds fetch function to submitForm function to send the form information to the marble api endpoint
- Adds an error message display element to the form under the Submit button
- Updates the config handlebars variable  and adds the "node_details" key to the handlebars variable that gets values from the "node_details" section of the config.toml file or config.default.toml file

mishaschwartz commented on templates/site/js/dataproduct-form.js:
```diff
@@ -1367,4 +1367,32 @@ async function submitForm(){
     submitObject["metadata"] = metadataObject;
     submitObject["path"] = linkedPathField.value;
+
+    var marbleAPIURL = "{{ configs['node_details']['marble_api_path'] }}";
```
```suggestion
    var marbleAPIURL = "{{ configs['node_details']['marble_api_path'] }}/v1/data-requests";
```

alexandercyu replied:
Changed the url to the suggested one.

mishaschwartz commented on templates/site/js/dataproduct-form.js:
```diff
@@ -1367,4 +1367,32 @@ async function submitForm(){
+
+        const result = await response.json();
+
+        if (!response.ok) {
```
typically if you have an if/else statement you don't negate the condition, just switch the blocks. It's easier to read.

mishaschwartz commented on templates/site/js/dataproduct-form.js:
```diff
@@ -1367,4 +1367,32 @@ async function submitForm(){
+            submitErrorElement.innerText = response.statusText;
+            throw new Error(`Response status: ${result.status_code}`);
+        }else{
+            submitErrorElement.innerText = result.detail;
```
The detail is present if there is an error, not if the request succeeds.

mishaschwartz commented on templates/site/js/dataproduct-form.js:
```diff
@@ -1367,4 +1367,32 @@ async function submitForm(){
     submitObject["metadata"] = metadataObject;
     submitObject["path"] = linkedPathField.value;
+
+    var marbleAPIURL = "{{ configs['node_details']['marble_api_path'] }}";
```
We discussed this already in #20 
Update your config file

alexandercyu commented on templates/site/js/dataproduct-form.js:
```diff
@@ -1367,4 +1367,32 @@ async function submitForm(){
+        const result = await response.json();
+
+        if (!response.ok) {
+            submitErrorElement.innerText = response.statusText;
```
Added check for details key and a generic error message for fallback condition.

alexandercyu commented:
- removed "metadataObject" object
- moved "variables", "extra_properties", "links", "temporal" from metadataObject to the parent submitObject hierarchy
- renamed "username" index to "user"
